### DISCUSSION TITLE
BadLinqExpressionException for nested lists after dotnet 8 migration

### DISCUSSION THREAD

**Discussion by xvrlsc** [JasperFx/marten#3388] [Q&A]
Hello! We are migrating our project from dotnet 6 to dotnet 8 so we upgraded Marten from v6.4.1 to v7.26.2. Following this migration we now have the following error:

```
BadLinqExpressionException
Marten does not (yet) support contains queries through collections of element type <dotnetObject>
```

And here is the concerned code that was working well before:
```c#
session.Query<DbObject>().Where(x => x.nestedReadOnlyCollection.ToList().Contains(someSearchedValue);
```

Thank you for your help on this.

**Comment by jeremydmiller** [comment]
@xvrlsc Need a little more information. Most likely, it's the usage of `ToList()` up above that's causing the issue. What type is `nestedReadOnlyCollection`?

Can you just use:

```csharp
session.Query<DbObject>().Where(x => x.nestedReadOnlyCollection.Contains(someSearchedValue);
```

**Reply by xvrlsc** [reply]
Hello @jeremydmiller. Sadly it does not work either without the ToList(). We have the same behaviour.
The type of the nestedReadOnlyCollection is System.Collections.Generic.IReadOnlyCollection<SomeComplexType>
